chain/client/src/sync.rs

```diff
@@ -685,24 +658,29 @@ impl StateSync {
                     shard_id,
                     hash,
                     need_header: true,
-                    parts_ranges: vec![],
+                    part_ids: vec![],
+                    num_parts: 0, /* we don't use it here */
                     target: possible_targets[thread_rng().gen_range(0, possible_targets.len())]
                         .clone(),
                 });
                 assert!(new_shard_sync_download.downloads[0].run_me);
                 new_shard_sync_download.downloads[0].run_me = false;
             }
             ShardSyncStatus::StateDownloadParts => {
-                let download_strategy =
-                    StateSyncStrategy::download_sqrt(&shard_sync_download.downloads);
-                self.apply_download_strategy(
-                    shard_id,
-                    hash,
-                    &possible_targets,
-                    download_strategy,
-                    &shard_sync_download,
-                    &mut new_shard_sync_download,
-                )?;
+                for (i, download) in new_shard_sync_download.downloads.iter().enumerate() {
+                    if download.run_me {
+                        self.network_adapter.send(NetworkRequests::StateRequest {
+                            shard_id,
+                            hash,
+                            need_header: false,
+                            part_ids: vec![i as u64],
+                            num_parts: new_shard_sync_download.downloads.len() as u64,
+                            target: possible_targets
+                                [thread_rng().gen_range(0, possible_targets.len())]
```

Comment:
This will panic if the node has no known peers https://docs.rs/rand/0.7.2/rand/trait.Rng.html#method.gen_range

Reply:
No, it cannot.
See the code above:
```
        if possible_targets.len() == 0 {
            return Ok(shard_sync_download);
        }
```